src/library/scala/util/Try.scala

```diff
@@ -278,4 +278,9 @@ final case class Success[+T](value: T) extends Try[T] {
   override def toEither: Either[Throwable, T] = Right(value)
   override def fold[U](fa: Throwable => U, fb: T => U): U =
     try { fb(value) } catch { case NonFatal(e) => fa(e) }
+
+}
+
+object Success {
+  class UnsatisfiedPredicateException[T] private[Success] (val value: T) extends RuntimeException("Predicate does not hold", null, false, false)
```

Comment:
Would it be useful to use the `value` in the exception message ? E.g. `s"Predicate does not hold for $value"` ?

Reply:
if you override `getMessage` instead, could you make the computation of the message lazy?